kratos/utilities/math_utils.h

```diff
@@ -62,180 +27,471 @@ SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+        }
+        else if (TDim == 3)
+        {
+            InputMatrixDet = InputMatrix(0, 0) * InputMatrix(1, 1) * InputMatrix(2, 2)
+                           + InputMatrix(1, 0) * InputMatrix(2, 1) * InputMatrix(0, 2)
+                           + InputMatrix(0, 1) * InputMatrix(1, 2) * InputMatrix(2, 0)
+                           - InputMatrix(2, 0) * InputMatrix(1, 1) * InputMatrix(0, 2)
+                           - InputMatrix(2, 1) * InputMatrix(1, 2) * InputMatrix(0, 0)
+                           - InputMatrix(1, 0) * InputMatrix(0, 1) * InputMatrix(2,2);
+        }
+        else
+        {
+            InputMatrixDet = (InputMatrix(0,0)*InputMatrix(1,1)*InputMatrix(2,2)*InputMatrix(3,3)) + (InputMatrix(0,0)*InputMatrix(1,2)*InputMatrix(2,3)*InputMatrix(3,1))
+                        + (InputMatrix(0,0)*InputMatrix(1,3)*InputMatrix(2,1)*InputMatrix(3,2)) + (InputMatrix(0,1)*InputMatrix(1,0)*InputMatrix(2,3)*InputMatrix(3,2))
+                        + (InputMatrix(0,1)*InputMatrix(1,2)*InputMatrix(2,0)*InputMatrix(3,3)) + (InputMatrix(0,1)*InputMatrix(1,3)*InputMatrix(2,2)*InputMatrix(3,0))
+                        + (InputMatrix(0,2)*InputMatrix(1,0)*InputMatrix(2,1)*InputMatrix(3,3)) + (InputMatrix(0,2)*InputMatrix(1,1)*InputMatrix(2,3)*InputMatrix(3,0))
+                        + (InputMatrix(0,2)*InputMatrix(1,3)*InputMatrix(2,0)*InputMatrix(3,1)) + (InputMatrix(0,3)*InputMatrix(1,0)*InputMatrix(2,2)*InputMatrix(3,1))
+                        + (InputMatrix(0,3)*InputMatrix(1,1)*InputMatrix(2,0)*InputMatrix(3,2)) + (InputMatrix(0,3)*InputMatrix(1,2)*InputMatrix(2,1)*InputMatrix(3,0))
+                        - (InputMatrix(0,0)*InputMatrix(1,1)*InputMatrix(2,3)*InputMatrix(3,2)) - (InputMatrix(0,0)*InputMatrix(1,2)*InputMatrix(2,1)*InputMatrix(3,3))
+                        - (InputMatrix(0,0)*InputMatrix(1,3)*InputMatrix(2,2)*InputMatrix(3,1)) - (InputMatrix(0,1)*InputMatrix(1,0)*InputMatrix(2,2)*InputMatrix(3,3))
+                        - (InputMatrix(0,1)*InputMatrix(1,2)*InputMatrix(2,3)*InputMatrix(3,0)) - (InputMatrix(0,1)*InputMatrix(1,3)*InputMatrix(2,0)*InputMatrix(3,2))
+                        - (InputMatrix(0,2)*InputMatrix(1,0)*InputMatrix(2,3)*InputMatrix(3,1)) - (InputMatrix(0,2)*InputMatrix(1,1)*InputMatrix(2,0)*InputMatrix(3,3))
+                        - (InputMatrix(0,2)*InputMatrix(1,3)*InputMatrix(2,1)*InputMatrix(3,0)) - (InputMatrix(0,3)*InputMatrix(1,0)*InputMatrix(2,1)*InputMatrix(3,2))
+                        - (InputMatrix(0,3)*InputMatrix(1,1)*InputMatrix(2,2)*InputMatrix(3,0)) - (InputMatrix(0,3)*InputMatrix(1,2)*InputMatrix(2,0)*InputMatrix(3,1));
+        }
+        
+        return InputMatrixDet;
+    }
+    
+    /**
+     * Calculates the inverse of a 2x2, 3x3 and 4x4 matrices (using bounded matrix for performance) 
+     * @param InputMatrix: The matrix to invert
+     * @return InvertMatrix: The inverted matrix
+     */
 
-        if(size==1)
```

Comment:
you can NOT change the API!!

the function InvertMatrix was accepting a matrix of variable size.
The new function you added accepts a matrix of fixed size.

while both functions may be useful you must preserve the original one.

also for the case of fixed-size matrices u should define 3 functions  accepting matrices of size 2*3, 3*3 etc, since this way the if can be avoided